Production_scripts/Social_Care/Care_Home.R

```diff
@@ -0,0 +1,416 @@
+  # For the records which still have multiple options (usually multiple names matched)
+  filter(n() == 1 | int_end(open_interval) == latest_close_date) %>%
+  filter(n() == 1 | match_mean == min_match_mean) %>%
+  ungroup() %>%
+  # Bring back to single record with no duplicates introduce by the lookup
+  distinct(ch_record_id, .keep_all = TRUE) %>%
+  # Replace the ch name with our best guess at the proper name from the lookup
+  mutate(
+    ch_name_old = ch_name,
+    ch_name = dplyr::if_else(is.na(ch_name_validated), ch_name, ch_name_validated)
+  ) %>%
+  # Bring back the records which had no postcode match
+  bind_rows(no_postcode_match)
+
+(check_names <- name_postcode_clean %>%
+  count(ch_name_old, ch_name, sort = TRUE))
+
+# Data Cleaning Care Home Data ---------------------------------------
+
+ch_data_clean <- name_postcode_clean %>%
+  # sort data
+  arrange(sending_location, social_care_id, ch_admission_date, period) %>%
+  mutate(
+    min_ch_provider = min(ch_provider),
+    max_ch_provider = max(ch_provider)
+  ) %>%
+  mutate(ch_provider = if_else(min_ch_provider != max_ch_provider, 6L, ch_provider)) %>%
+  select(-c(min_ch_provider, max_ch_provider)) %>%
+  # when multiple social_care_id from sending_location for single CHI
+  # replace social_care_id with latest
+  group_by(sending_location, chi) %>%
+  mutate(latest_sc_id = last(social_care_id)) %>%
+  # count changed social_care_id
+  mutate(
+    changed_sc_id = !is.na(chi) & social_care_id != latest_sc_id,
+    social_care_id = if_else(changed_sc_id, latest_sc_id, social_care_id)
+  ) %>%
+  ungroup() %>%
+  group_by(sending_location, social_care_id, ch_admission_date) %>%
+  # fill in nursing care provision when missing but present in the following entry
+  mutate(nursing_care_provision = na_if(nursing_care_provision, 9)) %>%
+  fill(nursing_care_provision, .direction = "downup") %>%
+  # tidy up ch_provider using 6 when disagreeing values
+  fill(ch_provider, .direction = "downup") %>%
+  # remove any duplicate records before merging for speed and simplicity
+  distinct() %>%
+  # counter for split episodes
+  mutate(
+    split_episode = replace_na(nursing_care_provision != lag(nursing_care_provision), TRUE),
+    split_episode_counter = cumsum(split_episode)
+  ) %>%
+  ungroup()
+
+
+# count changed social_care_id
+ch_data_clean %>% count(changed_sc_id)
+
+
+# Merge Records ---------------------------------------
+
+# to a single row per episode where admission the same
+ch_episode <- ch_data_clean %>%
+  # when nursing_care_provision is different on records within the episode, split the episode at this point
+  group_by(
+    chi,
+    sending_location,
+    social_care_id,
+    ch_admission_date,
+    nursing_care_provision,
+    split_episode_counter
+  ) %>%
+  summarise(
+    across(
+      c(
+        ch_discharge_date,
+        ch_provider,
+        record_date,
+        qtr_start,
+        sc_latest_submission,
```

Comment:
It seems that this sc_latest_submission column does not exist in ch_data_clean. So it will report an error when running this bit of script. Would it be an issue of the dataset?